packages/modules/devices/solax/solax_gen5/inverter.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+from typing import Dict, Union
+
+from dataclass_utils import dataclass_from_dict
+from modules.common import modbus
+from modules.common.abstract_device import AbstractInverter
+from modules.common.component_state import InverterState
+from modules.common.component_type import ComponentDescriptor
+from modules.common.fault_state import ComponentInfo, FaultState
+from modules.common.modbus import ModbusDataType
+from modules.common.store import get_inverter_value_store
+from modules.devices.solax.solax_gen5.config import SolaxGen5InverterSetup
+
+
+class SolaxGen5Inverter(AbstractInverter):
+    def __init__(self,
+                 device_id: int,
+                 component_config: Union[Dict, SolaxGen5InverterSetup],
+                 tcp_client: modbus.ModbusTcpClient_,
+                 modbus_id: int) -> None:
+        self.component_config = dataclass_from_dict(SolaxGen5InverterSetup, component_config)
+        self.__modbus_id = modbus_id
+        self.__tcp_client = tcp_client
+        self.store = get_inverter_value_store(self.component_config.id)
+        self.fault_state = FaultState(ComponentInfo.from_component_config(self.component_config))
+
+    def update(self) -> None:
+        with self.__tcp_client:
+            power_pv1 = self.__tcp_client.read_input_registers(10, ModbusDataType.UINT_16, unit=self.__modbus_id) * -1
+            power_pv2 = self.__tcp_client.read_input_registers(11, ModbusDataType.UINT_16, unit=self.__modbus_id) * -1
+            power_pv3 = self.__tcp_client.read_input_registers(292, ModbusDataType.UINT_16, unit=self.__modbus_id) * -1
+            power_temp = (power_pv1, power_pv2, power_pv3)
+            power = sum(power_temp)
+            exported = self.__tcp_client.read_input_registers(80, ModbusDataType.UINT_16, unit=self.__modbus_id) * 100
```

Comment:
Ist beim Solax Gen5 Register 80 der Gesamt-Ertrag oder der Tagesertrag?

Reply:
Tagesertrag
![grafik](https://github.com/user-attachments/assets/5162a407-4cac-4fbb-9d06-d80af9b3754d)

Reply:
Das ist der Gesamtertrag.

Reply:
Von der Regelung wird aktuell bei den WR und Speichern nur die Leistung verwendet. Du kannst aber z. B. die Ströme gerne mit auslesen. Die müssen auch nicht 1:1 zur Wirkleistung passen.

Reply:
Dann lass ich die Ströme mit drin.

Reply:
Viele Register stimmen mit dem Ursprungsmodul überein. Es liegen hier nur vereinzelte Registerabweichungen vor (bspw zusätzliches Erfassen von PV3 Power; SoC Register 302 statt 28; Auslesen der Gesamtzählerstände).
Es wäre besser das als zusätzliche Option zur Auswahl im bestehenden Modul anzubieten statt ein eigenes Modul zu erstellen (auch um redundanten Code zu vermeiden).

Reply:
@ndrsnhs Ist auch ein Ansatz. Wo kann man denn die UI für die Einstellungsseite in einem laufenden System zum testen anpassen?

Reply:
@rgrae81 
https://github.com/openWB/core/wiki/Einstellungs-Seite-erstellen
Das settings-Repo muss geforkt werden und dann die dortigen Änderungen kompilieren.